api/app/tests/apps/api/test_filescanner.py

```diff
@@ -0,0 +1,115 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 Gemeente Amsterdam
+"""
+Test uploads, specifically test the filetype sniffing. Note we are not testing
+the permissions system of SIA/Signalen, hence the usage of superusers to write
+to private endpoints
+"""
+import os
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+from signals.apps.signals.factories import SignalFactory
+from tests.test import SignalsBaseApiTestCase
+
+THIS_DIR = os.path.dirname(__file__)
+TEST_DATA_DIR = os.path.join(os.path.split(THIS_DIR)[0], 'signals', 'test-data')
+
+DOC_FILE = os.path.join(TEST_DATA_DIR, 'sia-ontwerp-testfile.doc')
+JPG_FILE = os.path.join(TEST_DATA_DIR, 'test.jpg')
+PNG_FILE = os.path.join(TEST_DATA_DIR, 'test.png')
+GIF_FILE = os.path.join(TEST_DATA_DIR, 'test.gif')
+
+ALLOWED = [
+    (GIF_FILE, 'image/gif'),
+    (JPG_FILE, 'image/jpeg'),
+    (PNG_FILE, 'image/png')
+]
+DISALLOWED = [(DOC_FILE, 'application/msword')]
+
+
+class TestUploadedFileScannerViaAPI(SignalsBaseApiTestCase):
```

Comment:
If you only want to test the scanner class this unit test is testing way too much. It is also testing the whole attachment endpoint. I think this test should only test the scanner class and nothing else. The API and upload of attachments is already tested in the `test_public_signal_endpoint.py` and `test_private_signal_endpoint.py` unit tests.

Reply:
I added tests at the services level